src/inmanta/server/services/orchestrationservice.py

```diff
@@ -936,6 +930,19 @@ async def release_version(
                     connection=connection,
                 )
 
+        increments: tuple[abc.Set[ResourceIdStr], abc.Set[ResourceIdStr]] = await self.resource_service._get_increment(
+            env, version_id
+        )
+
+        increment_ids, neg_increment = increments
+        await self.resource_service._mark_deployed(env, neg_increment, now, lambda x: True, version_id)
+
+        await model.update_fields(released=True, result=const.VersionState.deploying, connection=connection)
+
```

Comment:
Why did we move this call down? I find it a bit strange that we send resource_action_update calls with the action  `ResourceAction.deploy` when the state of the ConfigurationModel hasn't been set to deploying yet.

Reply:
This is the moment agents can start to pull in the resources (by setting `released=True`) so either all of this goes into a transaction or this one has to be at the end,...

Reply:
I see. I don't think we need a transaction here. But I would definitely add this explanation in the code as a comment.